File: src/main/java/com/itsci/mju/maebanjumpen/transaction/dto/TransactionDTO.java
package com.itsci.mju.maebanjumpen.transaction.dto;

import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.itsci.mju.maebanjumpen.partyrole.dto.MemberDTO;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class TransactionDTO {
    private Integer transactionId;
    private String transactionType;
    private Double transactionAmount;
    private LocalDateTime transactionDate;
    private String transactionStatus;

    @JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
    private MemberDTO member;

    private String prompayNumber;
    private String bankAccountNumber;
    private String bankAccountName;
    private LocalDateTime transactionApprovalDate;
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/transaction/dto/TransactionDTO.kt
package com.itsci.mju.maebanjumpen.transaction.dto

import com.fasterxml.jackson.annotation.JsonTypeInfo
import com.itsci.mju.maebanjumpen.partyrole.dto.MemberDTO
import java.time.LocalDateTime

data class TransactionDTO(
    var id: Long? = 0,
    var transactionType: String? = null,
    var transactionAmount: Double? = null,
    var transactionDate: LocalDateTime? = null,
    var transactionStatus: String? = null,
    @JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
    var member: MemberDTO? = null,
    var prompayNumber: String? = null,
    var bankAccountNumber: String? = null,
    var bankAccountName: String? = null,
    var transactionApprovalDate: LocalDateTime? = null
)

